alerting/provider/gitlab/gitlab.go

```diff
@@ -0,0 +1,147 @@
+package gitlab
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/TwiN/gatus/v5/alerting/alert"
+	"github.com/TwiN/gatus/v5/client"
+	"github.com/TwiN/gatus/v5/core"
+	"github.com/dchest/uniuri"
+)
+
+// AlertProvider is the configuration necessary for sending an alert using Gitlab
+type AlertProvider struct {
+	WebhookURL       string `yaml:"webhook-url"`       // The URL of the GitHub repository to create issues in
+	AuthorizationKey string `yaml:"authorization-key"` // Token requires at least RW on issues and RO on metadata
+
+	// DefaultAlert is the default alert configuration to use for endpoints with an alert of the appropriate type
+	DefaultAlert *alert.Alert `yaml:"default-alert,omitempty"`
+
+	// Severity can be one of: critical, high, medium, low, info, unknown. Defaults to critical
+	Severity string `yaml:"severity,omitempty"`
+	// MonitoringTool overrides the name sent to gitlab. Defaults to gatus
+	MonitoringTool string `yaml:"monitoring-tool,omitempty"`
+	// EnvironmentName is the name of the associated GitLab environment. Required to display alerts on a dashboard.
+	EnvironmentName string `yaml:"environment-name,omitempty"`
+	// Service affected. Defaults to endpoint display name
+	Service string `yaml:"service,omitempty"`
+}
+
+// IsValid returns whether the provider's configuration is valid
+func (provider *AlertProvider) IsValid() bool {
+	if len(provider.AuthorizationKey) == 0 || len(provider.WebhookURL) == 0 {
+		return false
+	}
+	// Validate format of the repository URL
+	_, err := url.Parse(provider.WebhookURL)
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+// Send creates an issue in the designed RepositoryURL if the resolved parameter passed is false,
+// or closes the relevant issue(s) if the resolved parameter passed is true.
+func (provider *AlertProvider) Send(endpoint *core.Endpoint, alert *alert.Alert, result *core.Result, resolved bool) error {
+	if len(alert.ResolveKey) == 0 {
+		alert.ResolveKey = uniuri.NewLen(uniuri.UUIDLen)
```

Comment:
I know it may sound a little nitpicky, but I would like to limit the number of 3rd party libraries as much as possible, unless they are absolutely necessary, and in this case, it seems like uniuri is used to generate a random string of character, and Gatus is already using a library to generate a random string of character & you could use `uuid.NewString()`.

For an example of how it is currently used in the code, see: https://github.com/TwiN/gatus/blob/3350e8144376c1fbe002469248af0427cfe985fa/security/oidc.go#L51

Reply:
> I know it may sound a little nitpicky

Not at all, I could have checked for existing libs. This is now fixed.